ui/vr_webgl.js

```diff
@@ -0,0 +1,611 @@
+    }
+    let currentHeight = this.video_.videoHeight;
+    if (!currentHeight) {
+      currentHeight = this.canvas_.scrollHeight;
+    }
+
+    if (this.previousCanvasWidth_ !== currentWidth ||
+        this.previousCanvasHeight_ !== currentHeight) {
+      this.canvas_.width = currentWidth;
+      this.canvas_.height = currentHeight;
+
+      this.previousCanvasWidth_ = currentWidth;
+      this.previousCanvasHeight_ = currentHeight;
+
+      const ratio = currentWidth / currentHeight;
+
+      this.projectionMatrix_ = shaka.ui.Matrix4x4.frustum(
+          this.projectionMatrix_, -ratio, ratio, -1, 1, scale, 50.5);
+
+      this.gl_.viewport(0, 0, currentWidth, currentHeight);
+    }
+  }
+
+  /**
+   * Rotate the view matrix global
+   *
+   * @param {!number} yaw Yaw.
+   * @param {!number} pitch Pitch.
+   * @param {!number} roll Roll.
+   */
+  rotateViewGlobal(yaw, pitch, roll) {
+    const pitchBoundary = 90.0 * Math.PI / 180;
+    const matrix = this.viewMatrix_;
+
+    // Rotate global axis
+    shaka.ui.Matrix4x4.rotateY(matrix, matrix, yaw);
+
+    // Variable to limit the pitch movement
+    this.positionY_ += pitch;
+
+    if (this.positionY_ < pitchBoundary &&
+      this.positionY_ > -pitchBoundary) {
+      const out = shaka.ui.Matrix4x4.create();
+      shaka.ui.Matrix4x4.rotateX(out, shaka.ui.Matrix4x4.create(), -1 * pitch);
+      // Rotate local axis
+      shaka.ui.Matrix4x4.multiply(matrix, out, matrix);
+    } else {
+    // Doing this we restart the value to the previous position,
+    // to not mantain a value over 90º or under -90º.
+      this.positionY_ -= pitch;
+    }
+
+    const out2 = shaka.ui.Matrix4x4.create();
+    shaka.ui.Matrix4x4.rotateZ(out2, shaka.ui.Matrix4x4.create(), roll);
+
+    // Rotate local axis
+    shaka.ui.Matrix4x4.multiply(matrix, out2, matrix);
+
+    this.renderGL_(false);
+  }
+
+  /**
+   * @param {number} amount
+   */
+  zoom(amount) {
+    const zoomMin = 20;
+    const zoomMax = 100;
+    amount /= 50;
+    if (this.fieldOfView_ >= zoomMin && this.fieldOfView_ <= zoomMax) {
+      this.fieldOfView_ += amount;
+    }
+    if (this.fieldOfView_ < zoomMin) {
+      this.fieldOfView_ = zoomMin;
+    } else if (this.fieldOfView_ > zoomMax) {
+      this.fieldOfView_ = zoomMax;
+    }
+    this.renderGL_(false);
+  }
+
+  /**
+   * @return {number}
+   */
+  getFieldView() {
+    return this.fieldOfView_;
+  }
+
+  /**
+   * @param {number} fieldView
+   */
+  setFieldView(fieldView) {
+    this.fieldOfView_ = fieldView;
+    this.renderGL_(false);
+  }
+
+  /**
+   * @return {number}
+   */
+  getNorth() {
+    shaka.ui.Matrix4x4.getRotation(this.currentQuaternion_, this.viewMatrix_);
+
+    const angles = this.toEulerAngles_(this.currentQuaternion_);
+
+    let result = ((angles.yaw * 180) / Math.PI);
+    if (result < 0) {
+      result = 360 + result;
+    }
+    return result;
+  }
+
+  /**
+   * @param {boolean=} firstTime
+   */
+  reset(firstTime = true) {
```

Comment:
Perhaps I missed it... what calls the `reset` method initially, to start the animation? It doesn't seem to be called by the VR manager.

Reply:
It is to reset the position to the initial position, I forgot to add it in the controls.